wgpu-core/src/command/bundle.rs

```diff
@@ -688,7 +731,9 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
                             .buffers
                             .use_extend(&*buffer_guard, buffer_id, (), BufferUse::INDEX)
                             .unwrap();
-                        assert!(buffer.usage.contains(wgt::BufferUsage::INDEX), "An invalid setIndexBuffer call has been made. The buffer usage is {:?} which does not contain required usage INDEX", buffer.usage);
+                        if !buffer.usage.contains(wgt::BufferUsage::INDEX) {
+                            return Err(RenderCommandError::InvalidBufferUsage);
```

Comment:
Maybe `MissingBufferUsage(wgt::BufferUsage::INDEX)`?

Reply:
For things that have a log message, I would prefer that the original message gets `log::error!`'d to the logs so that users can still see the full message written in plain english.